File: Lainnya/movie-catalogue-jetpack-2-main/movie-catalogue-jetpack-2-main/app/src/main/java/com/mramirid/moviecatalogue/ui/movies/MoviesViewModel.java
package com.mramirid.moviecatalogue.ui.movies;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.mramirid.moviecatalogue.data.entity.ItemEntity;
import com.mramirid.moviecatalogue.data.source.MovieCatalogueRepository;

import java.util.ArrayList;

public class MoviesViewModel extends ViewModel {

	private MovieCatalogueRepository movieCatalogueRepository;

	public MoviesViewModel(MovieCatalogueRepository movieCatalogueRepository) {
		this.movieCatalogueRepository = movieCatalogueRepository;
	}

	LiveData<ArrayList<ItemEntity>> getMovies() {
		return movieCatalogueRepository.getMovies();
	}
}

File: Lainnya/cielo/JetpackSubmission3v2/app/src/main/java/com/gracielo/jetpacksubmission3v2/Movies/MoviesViewModel.kt
package com.gracielo.jetpacksubmission3v2.Movies

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.paging.PagedList
import com.gracielo.jetpacksubmission3v2.Data.Local.Entity.MovieEntity
import com.gracielo.jetpacksubmission3v2.Data.Source.MovieCatalogueRepository
import com.gracielo.jetpacksubmission3v2.vo.Resource
import java.util.*

class MoviesViewModel(movieCatalogueRepository: MovieCatalogueRepository) : ViewModel() {
    private val movieCatalogueRepository: MovieCatalogueRepository

    init {
        this.movieCatalogueRepository = movieCatalogueRepository
    }

    fun getMovies(): LiveData<Resource<PagedList<MovieEntity>>> {
        return movieCatalogueRepository.getMovies()
    }

}